querier/querier.go

```diff
@@ -206,6 +201,97 @@ func (qm MergeQuerier) Close() error {
 	return nil
 }
 
+// RemoteReadHandler handles Prometheus remote read requests.
+func (qm MergeQuerier) RemoteReadHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var req cortex.ReadRequest
+	if err := distributor.ParseProtoRequest(ctx, w, r, &req, true); err != nil {
```

Comment:
Ah just noticed that `ParseProtoRequest` has:

```go
instrument.TimeRequestHistogram(ctx, "Distributor.PushHandler[decompress]", ...
```

What do you say - maybe just rename the string to `"util.ParseProtoRequest[decompress]"`?